### ISSUE TITLE
Method signature mismatch

### ISSUE THREAD

**Issue by FlyingDR** [thephpleague/oauth2-google#4]
Current version of ```Google``` provider class is broken when used together with ```league/oauth2-client``` version ```1.0.0-beta1```. Method signature in [AbstractProvider](https://github.com/thephpleague/oauth2-client/blob/1.0.0-beta1/src/Provider/AbstractProvider.php#L646) is:
```php
abstract protected function checkResponse(ResponseInterface $response, $data);
```
but this method in [current version](https://github.com/thephpleague/oauth2-google/blob/6953874780cce2f54f689f355a2109891696dc08/src/Provider/Google.php#L70) of ```Google``` provider is:
```php
protected function checkResponse(array $response);
```
and it causes fatal error.

**Comment by shadowhand** [comment]
Similar issue reported in #3. I'll merge them all into a single issue.

**Comment by shadowhand** [comment]
Merged into #5.
